src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -752,6 +791,96 @@ protected function compileEndpush($expression)
     }
 
     /**
+     * Compile the inject annotations into valid PHP.
+     *
+     * @param  string[] $segments
+     *
+     * @return string
+     */
+    protected function compileInjectAnnotation(array $segments)
+    {
+        list($injectClass, $variableName) = $segments;
+
+        return '<?php '.$variableName.' = app('.$injectClass.'::class); ?>';
```

Comment:
You could do `@inject('auth', 'auth.driver')` (service location), but this is not possible with the neew approach.